src/authenticateRequest.ts

```diff
@@ -1,169 +1,91 @@
-import { AuthenticationError, ForbiddenError } from 'apollo-server-express'
-import { Request } from 'express'
-import { Maybe } from 'graphql/jsutils/Maybe'
-import jwt, { GetPublicKeyOrSecret } from 'jsonwebtoken'
-import jwksClient from 'jwks-rsa'
-import { omit } from 'lodash'
+import * as crypto from 'crypto'
+import { CookieOptions } from 'express'
+import { Strategy as CookieStrategy } from 'passport-cookie'
 import UserAccount from './models/user_account'
 
-// These values are from the perspective of Auth0.
-// audience is the Shipment Tracker API, while issuer is the
-// distributeaid tenant in Auth0 that owns the API.
-// jwksUri is where we can get the signing key from Auth0.
-const audience: Maybe<string> =
-  process.env.API_IDENTIFIER || 'https://da-shipping-tracker-dev'
-const issuer = process.env.AUTH0_ISSUER || 'https://distributeaid.eu.auth0.com/'
-const jwksUri =
-  process.env.AUTH0_JWKS_URI ||
-  'https://distributeaid.eu.auth0.com/.well-known/jwks.json'
-
-const client = jwksClient({
-  jwksUri,
-})
-
-type SigningKey = {
-  publicKey?: string
-  rsaPublicKey?: string
-}
-
-const getAuth0SigningKey: GetPublicKeyOrSecret = (header, callback) => {
-  if (!header?.kid) {
-    return callback('Header is missing kid')
-  }
-
-  client.getSigningKey(header.kid, (error, key: SigningKey) => {
-    if (error != null) callback(error)
-
-    const signingKey: string | undefined = key.publicKey || key.rsaPublicKey
-
-    callback(null, signingKey)
-  })
-}
-
-type AuthResult = {
-  decoded: object | null
-  error?: object
-}
-
-const validateJwt = async (token: string): Promise<AuthResult> => {
-  return new Promise((resolve, reject) => {
-    if (!token.startsWith('Bearer ')) {
-      return reject(new Error('Auth token is malformed'))
-    }
-
-    const bearerToken = token.split(' ')[1]
-
-    jwt.verify(
-      bearerToken,
-      getAuth0SigningKey,
-      {
-        audience,
-        issuer,
-        algorithms: ['RS256'],
-      },
-      (error, decoded) => {
-        resolve({ decoded: decoded!, error: error! })
-      },
-    )
-  })
-}
-
-const ADMIN_ROLE = 'admin'
-const ROLES_KEY = 'http://id.distributeaid.org/role'
-
-type CommonAuthClaims = {
-  iss: string
-  sub: string
-  aud: string[]
-  iat: number
-  exp: number
-  azp: string
-  scope: string
-  permissions: string[]
-}
-
-export type RawAuthClaims = CommonAuthClaims & {
-  [ROLES_KEY]: string[]
+type AuthCookiePayload = {
+  /** user ID */
+  i: number
+  /** user is admin */
+  a: boolean
+  /** user hash */
+  c: string
```

Comment:
**question (non-blocking):** are those fields named that way to match the way `passport-cookie` formats its data?

Reply:
No, but we encode this in the cookie sent to the browser, and there is a size limit, so I want to keep it as small as possible (while still using JSON).